### PR TITLE
Add undocumented (ie, unsupported) config option (DI config only) to disable forcing range archiving on browser request.

### PR DESCRIPTION
**Pull request by diosmosis**
As title. From pro request on slack.

If in the future we decide there's a better way to do this, we can remove the option w/o any issue. If we decide it's a good thing, we can add an INI config option.

### CODE REVIEW

**Comment by tsteur** [comment]
@diosmosis I'm curious how PRO would change this config value when not adding to INI? By using a plugin? A change to this file would be overwritten on an update. 

BTW: Is there maybe a `&& !SettingsServer::isArchivePhpTriggered()` needed? Not sure if we have tests for archiving range dates via `core:archive` etc. If not, a test would be nice.

> If in the future we decide there's a better way to do this, we can remove the option w/o any issue. If we decide it's a good thing, we can add an INI config option.

Maybe also have a look at https://github.com/piwik/piwik/issues/7095#issuecomment-71907932 we need to clean up this mess. Don't have to do it in this PR though. Just FYI

**Comment by diosmosis** [comment]
> @diosmosis I'm curious how PRO would change this config value when not adding to INI? By using a plugin? A change to this file would be overwritten on an update. 

If they add the setting to a config/config.php file it should override the global config, see:
https://github.com/piwik/piwik/blob/master/core/Container/ContainerFactory.php#L65

That said, it would be simpler to just not put a default value in global.ini.php and handle the case where `empty(Config::getInstance()->General[...]) === true`.

> BTW: Is there maybe a && !SettingsServer::isArchivePhpTriggered() needed? Not sure if we have tests for archiving range dates via core:archive etc. If not, a test would be nice.

I don't think archiving ranges via core:archive is supported? I remember this is something that is being worked on but I don't think it's merged yet (haven't checked).

That said, it shouldn't be needed.

**Comment by tsteur** [comment]
> setting to a config/config.php file it should override the global config

Sweet, didn't know that

> I don't think archiving ranges via core:archive is supported? I remember this is something that is being worked on but I don't think it's merged yet (haven't checked).

I thought that's why one wants to disable it via browser? If it isn't pre-archived via `core:archive` and is not processed via browser then one could simply disable range dates or not? Isn't there a config entry?

**Comment by diosmosis** [comment]
> I thought that's why one wants to disable it via browser? If it isn't pre-archived via core:archive and is not processed via browser then one could simply disable range dates or not? Isn't there a config entry? 

Ok, from the slack chat, @mgazdzik is working on a command for pre-processing ranges. I'll add a test for archiving ranges via CLI.

File: core/ArchiveProcessor/Rules.php [hunk 1]
```diff
@@ -229,13 +230,19 @@ public static function getTodayArchiveTimeToLiveDefault()
     public static function isArchivingDisabledFor(array $idSites, Segment $segment, $periodLabel)
     {
         if ($periodLabel == 'range') {
-            return false;
+            if (StaticContainer::get('archiving.range.force_on_browser_request') !== false) {
+                return false;
+            } else {
+                Log::verbose("Not forcing archiving for range period.");
```
Comment (mnapoli) [inline core/ArchiveProcessor/Rules.php]: By the way in PSR-3 the log level "verbose" doesn't exist, I aliased it to "debug". So using verbose is the same as using debug.

**Comment by tsteur** [comment]
As long as it works I'm happy with it :) Just out of curiosity, is there a reason why you know longer use DI but config? I presume PRO will set a value then in config and they can maybe set it per instance / hostname this way?

**Comment by diosmosis** [comment]
> Just out of curiosity, is there a reason why you know longer use DI but config?

Using DI config was an experiment. It might be a good idea, but since there was a simpler and more standard way to accomplish the same thing I went w/ that. Principle of least surprise and all that.

**Comment by tsteur** [comment]
sweet :)

**Comment by mattab** [comment]
The new setting looks good! 

Feedback

 *   I think all settings should be documented in global.ini.php simply for us already to remember what they do, maybe we could add it around `browser_archiving_disabled_enforce`

**Pull request merged**